model/infinispan/src/main/java/org/keycloak/models/sessions/infinispan/InfinispanUserSessionProvider.java

```diff
@@ -341,28 +343,39 @@ private UserSessionEntity getUserSessionEntity(RealmModel realm, String id, bool
 
     protected Stream<UserSessionModel> getUserSessionsStream(RealmModel realm, UserSessionPredicate predicate, boolean offline) {
 
-        if (offline && loadOfflineSessionsStatsFromDatabase) {
+        if (offline && loadOfflineSessionsFromDatabase) {
 
             // fetch the offline user-sessions from the persistence provider
             UserSessionPersisterProvider persister = session.getProvider(UserSessionPersisterProvider.class);
 
-            UserModel user = session.users().getUserById(realm, predicate.getUserId());
-            if (user != null) {
-                return persister.loadUserSessionsStream(realm, user, offline, 0, null);
+            if (predicate.getUserId() != null) {
+                UserModel user = session.users().getUserById(realm, predicate.getUserId());
+                if (user != null) {
+                    return persister.loadUserSessionsStream(realm, user, true, 0, null);
+                }
+            }
+
+            if (predicate.getBrokerUserId() != null) {
+                String[] idpAliasSessionId = predicate.getBrokerUserId().split("\\.");
+
+                Map<String, String> attributes = new HashMap<>();
+                attributes.put(UserModel.IDP_ALIAS, idpAliasSessionId[0]);
+                attributes.put(UserModel.IDP_USER_ID, idpAliasSessionId[1]);
+
+                UserProvider userProvider = session.getProvider(UserProvider.class);
+                UserModel userModel = userProvider.searchForUserStream(realm, attributes, 0, null).findFirst().orElse(null);
+                return userModel != null ?
+                        persister.loadUserSessionsStream(realm, userModel, true, 0, null) :
+                        Stream.empty();
             }
 
             if (predicate.getBrokerSessionId() != null) {
                 // TODO add support for offline user-session lookup by brokerSessionId
                 // currently it is not possible to access the brokerSessionId in offline user-session in a database agnostic way
-                throw new ModelException("Dynamic database lookup for offline user-sessions by brokerSessionId is currently only supported for preloaded sessions.");
+                throw new ModelException("Dynamic database lookup for offline user-sessions by broker session ID is currently only supported for preloaded sessions. " +
```

Comment:
It shouldn't happen with the current code-base. Only usage was in the backChannelLogout but that was changed. But still it can probably happen if someone uses some custom code.